### PR TITLE
OSSFuzz Bug Fix 

### PR DESCRIPTION
**Pull request by ennamarie19**
This pull request fixes a crash encountered during OSSFuzz continuous fuzzing. Previously, an InvalidTimeError exception in pytz's UTC offset when processing 01-01-1970 0400 would cause parsing to exit prematurely. This fix catches the exception and resolves dateobj_time the secondary way. 

Additionally, to detect bugs prior to them being merged into the codebase, the CIFuzz.yml job that comes with OSSFuzz is proposed in this PR. With this, each PR will be fuzz-tested as part of the CI workflow. 

Thank you for your time and your review!

### CODE REVIEW

**Comment by Gallaecio** [comment]
Could we include a test case for this scenario as well?

**Comment by capuanob** [comment]
@Gallaecio Good evening, we just pushed a test case to go with the change. It utilizes the same settings and args that caused the bug during fuzzing to ensure that this bug is not reintroduced down the line.

Thank you!

File: dateparser/parser.py [hunk 1]
```diff
@@ -564,9 +564,19 @@ def _correct_for_time_frame(self, dateobj, tz):
...
 
+            dateobj_time = None
+            if tz:
+                try:
+                    dateobj_time = (dateobj - tz.utcoffset(dateobj)).time()
+                except pytz.InvalidTimeError:
+                    pass
+
+            if not dateobj_time:
+                dateobj_time = dateobj.time()
```
Comment (Gallaecio) [inline dateparser/parser.py:578]: This code looks like a no-op at the moment, it defines a `dateobj_time` variable that is never used.

**Comment by ennamarie19** [comment]
@Gallaecio It looks like someone updated the same function and merged to main since this fix, which caused a merge conflict that was not resolved correctly. This latest commit actually handles the bug

**Review by Gallaecio — APPROVED** [review]
Thanks!

**Pull request merged**